From c47584ca7b3a8685e8e0110171b174566685794a Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Fri, 22 Mar 2019 15:05:13 -0700
Subject: [PATCH 0577/1851] v4l: xilinx: vip: Add YUV 10bit media bus formats

commit 556d310d878d79baa5b167233ca72a1ccb610e72 from
https://github.com/Xilinx/linux-xlnx.git

This patch introduces separate media bus formats for each 10bit pixel
formats. This way v4l2 subdev applications can differentiate between
8bit and 10bit formats.

	YUV420 10bit -> MEDIA_BUS_FMT_VYYUYY10_4X20
	YUV422 10bit -> MEDIA_BUS_FMT_UYVY10_1X20
	YUV444 10bit -> MEDIA_BUS_FMT_VUY10_1X30

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vip.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vip.c b/drivers/media/platform/xilinx/xilinx-vip.c
index 1939ec91231f..c80e2ec59cf9 100644
--- a/drivers/media/platform/xilinx/xilinx-vip.c
+++ b/drivers/media/platform/xilinx/xilinx-vip.c
@@ -28,9 +28,9 @@ static const struct xvip_video_format xvip_video_formats[] = {
 	  1, 12, V4L2_PIX_FMT_NV12, 2, 1, 1, 2 },
 	{ XVIP_VF_YUV_420, 8, NULL, MEDIA_BUS_FMT_VYYUYY8_1X24,
 	  1, 12, V4L2_PIX_FMT_NV12M, 2, 2, 1, 2 },
-	{ XVIP_VF_YUV_420, 10, NULL, MEDIA_BUS_FMT_VYYUYY8_1X24,
+	{ XVIP_VF_YUV_420, 10, NULL, MEDIA_BUS_FMT_VYYUYY10_4X20,
 	  1, 12, V4L2_PIX_FMT_XV15, 2, 1, 2, 2 },
-	{ XVIP_VF_YUV_420, 10, NULL, MEDIA_BUS_FMT_VYYUYY8_1X24,
+	{ XVIP_VF_YUV_420, 10, NULL, MEDIA_BUS_FMT_VYYUYY10_4X20,
 	  1, 12, V4L2_PIX_FMT_XV15M, 2, 2, 1, 2 },
 	{ XVIP_VF_YUV_420, 12, NULL, MEDIA_BUS_FMT_UYYVYY12_4X24,
 	  1, 12, V4L2_PIX_FMT_X012, 2, 1, 2, 2 },
@@ -48,9 +48,9 @@ static const struct xvip_video_format xvip_video_formats[] = {
 	  1, 16, V4L2_PIX_FMT_NV16, 2, 1, 1, 1 },
 	{ XVIP_VF_YUV_422, 8, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
 	  1, 16, V4L2_PIX_FMT_NV16M, 2, 2, 1, 1 },
-	{ XVIP_VF_YUV_422, 10, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
+	{ XVIP_VF_YUV_422, 10, NULL, MEDIA_BUS_FMT_UYVY10_1X20,
 	  1, 16, V4L2_PIX_FMT_XV20, 2, 1, 2, 1 },
-	{ XVIP_VF_YUV_422, 10, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
+	{ XVIP_VF_YUV_422, 10, NULL, MEDIA_BUS_FMT_UYVY10_1X20,
 	  1, 16, V4L2_PIX_FMT_XV20M, 2, 2, 1, 1 },
 	{ XVIP_VF_YUV_422, 12, NULL, MEDIA_BUS_FMT_UYVY12_1X24,
 	  1, 16, V4L2_PIX_FMT_X212, 2, 1, 2, 1 },
@@ -64,7 +64,7 @@ static const struct xvip_video_format xvip_video_formats[] = {
 	  3, 24, V4L2_PIX_FMT_VUY24, 1, 1, 1, 1 },
 	{ XVIP_VF_YUVX, 8, NULL, MEDIA_BUS_FMT_VUY8_1X24,
 	  4, 32, V4L2_PIX_FMT_XVUY32, 1, 1, 1, 1 },
-	{ XVIP_VF_YUVX, 10, NULL, MEDIA_BUS_FMT_VUY8_1X24,
+	{ XVIP_VF_YUVX, 10, NULL, MEDIA_BUS_FMT_VUY10_1X30,
 	  4, 32, V4L2_PIX_FMT_XVUY10, 1, 1, 1, 1 },
 	{ XVIP_VF_RBG, 8, NULL, MEDIA_BUS_FMT_RBG888_1X24,
 	  3, 24, V4L2_PIX_FMT_BGR24, 1, 1, 1, 1 },
-- 
2.31.1

